nemo_skills/pipeline/prepare_data.py

```diff
@@ -0,0 +1,127 @@
+    mount_paths: str = typer.Option(None, help="Comma separated list of paths to mount"),
+    run_after: List[str] = typer.Option(None, help="List of expnames that this job depends on before starting"),
+    reuse_code: bool = typer.Option(True, help="Whether to reuse code from previous experiments"),
+    reuse_code_exp: str = typer.Option(None, help="Experiment to reuse code from"),
+    config_dir: str = typer.Option(None, help="Custom cluster config directory"),
+    with_sandbox: bool = typer.Option(False, help="Start a sandbox container alongside"),
+    log_dir: str = typer.Option(None, help="Custom location for slurm logs"),
+    exclusive: bool = typer.Option(True, "--not_exclusive", help="Disable --exclusive flag"),
+    check_mounted_paths: bool = typer.Option(False, help="Check mounted paths availability"),
+):
+    """Prepare datasets by running python -m nemo_skills.dataset.prepare.
+
+    Run `python -m nemo_skills.dataset.prepare --help` to see other supported arguments.
+    """
+    setup_logging(disable_hydra_logs=False, use_rich=True)
+    extra_arguments = f'{" ".join(ctx.args)}'
+    command = f"python -m nemo_skills.dataset.prepare {extra_arguments}"
+    if data_dir:
+        command += f" && mkdir -p {data_dir} && cp -r /nemo_run/code/nemo_skills/dataset/* {data_dir}"
+
+    cluster_config = get_cluster_config(cluster, config_dir=config_dir)
+    if cluster_config['executor'] == 'local' and not data_dir:
+        # in this case we need to put the results in the current folder
+        # if we use container, it will mess up permissions, so as a workaround
+        # setting executore to none
+        cluster_config['executor'] = 'none'
+
+    if cluster_config['executor'] == 'slurm' and not data_dir:
+        raise ValueError(
+            "Data directory is required to be specified when using slurm executor. "
+            "Please provide --data_dir argument."
+        )
+
+    log_dir = log_dir or data_dir
+
+    # we already captured extra arguments
+    ctx.args = []
+
+    if data_dir:
+        env_vars = get_env_variables(cluster_config)
+        data_dir_env_var = env_vars.get("NEMO_SKILLS_DATA_DIR") or os.environ.get("NEMO_SKILLS_DATA_DIR")
+        if data_dir_env_var != data_dir:
+            LOG.warning(
+                f"NEMO_SKILLS_DATA_DIR environment variable is set to {data_dir_env_var}, "
+                f"but you provided --data_dir={data_dir}. "
+                f"Make sure to set NEMO_SKILLS_DATA_DIR={data_dir} in your environment "
+                f"or pass it as an argument to `ns eval`."
+            )
+        if data_dir_env_var is None:
+            LOG.warning(
+                f"NEMO_SKILLS_DATA_DIR environment variable is not set. "
+                f"You might want to set it as NEMO_SKILLS_DATA_DIR={data_dir} "
+                f"to avoid always specifying it as a parameter to `ns eval`."
+            )
+        # TODO: automatically add it to cluster config based on user prompt?
```

Comment:
That would be nice. You can use is_mounted() and get_mounted() to test if its already in mount paths before adding it.